packages/core/src/render3/query.ts

```diff
@@ -489,50 +473,27 @@ function viewQueryInternal<T>(
  *
  * @param directiveIndex Current directive index
  * @param predicate The type for which the query will search
- * @param descend Whether or not to descend into children
+ * @param flags Flags associated with the query
  * @param read What to save in the query
  * @returns QueryList<T>
  *
  * @codeGenApi
  */
 export function ɵɵcontentQuery<T>(
-    directiveIndex: number, predicate: Type<any>|InjectionToken<unknown>|string[], descend: boolean,
-    read?: any): void {
-  contentQueryInternal(
-      getTView(), getLView(), predicate, descend, read, false, getCurrentTNode()!, directiveIndex);
-}
-
-/**
- * Registers a QueryList, associated with a static content query, for later refresh
- * (part of a view refresh).
- *
- * @param directiveIndex Current directive index
- * @param predicate The type for which the query will search
- * @param descend Whether or not to descend into children
- * @param read What to save in the query
- * @returns QueryList<T>
- *
- * @codeGenApi
- */
-export function ɵɵstaticContentQuery<T>(
-    directiveIndex: number, predicate: Type<any>|InjectionToken<unknown>|string[], descend: boolean,
-    read?: any): void {
-  contentQueryInternal(
-      getTView(), getLView(), predicate, descend, read, true, getCurrentTNode()!, directiveIndex);
-}
-
-function contentQueryInternal<T>(
-    tView: TView, lView: LView, predicate: Type<any>|InjectionToken<unknown>|string[],
-    descend: boolean, read: any, isStatic: boolean, tNode: TNode, directiveIndex: number): void {
+    directiveIndex: number, predicate: Type<any>|InjectionToken<unknown>|string[],
+    flags: QueryFlags, read?: any): void {
+  ngDevMode && assertNumber(flags, 'Expecting flags');
+  const tView = getTView();
   if (tView.firstCreatePass) {
-    createTQuery(tView, new TQueryMetadata_(predicate, descend, isStatic, read), tNode.index);
+    const tNode = getCurrentTNode()!;
+    createTQuery(tView, new TQueryMetadata_(predicate, flags, read), tNode.index);
     saveContentQueryAndDirectiveIndex(tView, directiveIndex);
-    if (isStatic) {
+    if (flags & QueryFlags.isStatic) {
```

Comment:
Should this be similar to other places?
```suggestion
    if ((flags & QueryFlags.isStatic) === QueryFlags.isStatic) {
```

Reply:
done